lib/gis_robot_suite/structural_file_builder.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module GisRobotSuite
+  # Builds cocina file metadata
+  class StructuralFileBuilder
+    def self.build(*args)
+      new(*args).build
+    end
+
+    # @param [Hash<Symbol,Assembly::ObjectFile>] objects
+    def initialize(assembly_objectfile, access, version)
+      @assembly_objectfile = assembly_objectfile
+      @access = access
+      @version = version
+    end
+
+    attr_reader :assembly_objectfile, :access, :version
+
+    def build
+      Cocina::Models::File.new(file_attributes)
+    end
+
+    private
+
+    def file_attributes
+      {
+        hasMimeType: mimetype,
+        administrative: { publish: true, shelve: true, sdrPreserve: roletype_for_file == 'master' },
+        access: access,
+        externalIdentifier: "https://cocina.sul.stanford.edu/file/#{SecureRandom.uuid}",
+        type: Cocina::Models::ObjectType.file,
+        version: version,
+        filename: assembly_objectfile.filename,
+        label: assembly_objectfile.filename,
+        size: assembly_objectfile.filesize,
+        use: roletype_for_file || 'master',
```

Comment:
Can this "|| master" be moved down to roletype_for_file?

Reply:
probably?  will try...

Reply:
oh, right, i'd noticed this when i was first reviewing the PR: i think that'd change the logic, since the `roletype_for_file == 'master'` for the `sdrPreserve` attribute (above on line 28) will always evaluate to `false` for anything that's not `mimetype == 'application/zip'` (because `roletype_for_file` will be `nil`).  but anything that's not `mimetype == 'application/zip'` will end up getting a `use` of `'master'`?  if i push the `|| 'master'` into `roletype_for_file`, i get a test failure (`spec/unit/gis_robot_suite/structural_metadata_builder_spec.rb:87`, an `sdrPreserve` that was `false` becomes `true`).

so maybe this was intentional, if not very clear in conveying the intent?

maybe separate `preserve?` and `roletype_for_file` methods are the way to go?

Reply:
Ahh, I see what you mean. I'd agree with you that something that better shows intent would be preferable.